File: Ruby/problem_9.rb
require 'pry'

(1...500).each do |a|
  b = a + 1
  loop do
    squared_sum = a**2 + b**2
    c = Math.sqrt(squared_sum)
    if a + b + c == 1000
      # overly fancy just to use inject
      puts [a, b, c].inject(1) { |prod, val| prod *= val }
    end
    break if a + b + c > 1000
    b += 1
  end
end

File: JS/problem9.js
var a = 1, b, squaredSum,
    c = 0;

(function() {
  for (; a < 500; a++) {
    b = a + 1;
    for (; a + b + c <= 1000; b++) {
      squaredSum = Math.pow(a,2) + Math.pow(b,2);
      c = Math.sqrt(squaredSum);
      if (a + b + c === 1000) {
        console.log(a*b*c);
      }
    }
  }
})();
